app/models/submission.rb

```diff
@@ -344,6 +346,16 @@ def can_view_plagiarism_report(type, user, session)
     )
   end
 
+  def user_can_peer_review?(user)
+    allowed = false
+    if assignment.peer_reviews && self.assigned_assessments
+      self.assigned_assessments.each do |review|
+        allowed = allowed || user.id == review.user_id
```

Comment:
I'm not sure I'm following what you're trying to do here.

If there are multiple assigned assessments, you'll just keep reassigning allowed.  Perhaps you want the .any? method here instead of .each?

Also, it looks like you really want allowed ||= user.id == review.user_id.  But again, I think the .any? method is more of what you're looking for.